testing/e2e/skr-tester/pkg/keb/client.go

```diff
@@ -0,0 +1,341 @@
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", errors.New("failed to get token")
+	}
+
+	var result map[string]interface{}
+	json.NewDecoder(resp.Body).Decode(&result)
+	o.Token = result["access_token"].(string)
+	o.Expiry = time.Now().Add(time.Duration(result["expires_in"].(float64)) * time.Second)
+
+	return o.Token, nil
+}
+
+func (c *KEBClient) BuildRequest(payload interface{}, endpoint, verb string) (*http.Request, error) {
+	token, err := c.Token.GetToken(SCOPES)
+	if err != nil {
+		return nil, err
+	}
+	platformRegion := c.GetPlatformRegion()
+	url := fmt.Sprintf("https://kyma-env-broker.%s/oauth/%sv2/%s", c.Host, platformRegion, endpoint)
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest(verb, url, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("X-Broker-API-Version", "2.14")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+	req.Header.Set("Content-Type", "application/json")
+	return req, nil
+}
+
+func (c *KEBClient) BuildRequestWithoutToken(payload interface{}, endpoint, verb string) (*http.Request, error) {
+	url := fmt.Sprintf("https://kyma-env-broker.%s/oauth/v2/%s", c.Host, endpoint)
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest(verb, url, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("X-Broker-API-Version", "2.14")
+	req.Header.Set("Content-Type", "application/json")
+	return req, nil
+}
+
+func (c *KEBClient) CallKEB(payload interface{}, endpoint, verb string) (map[string]interface{}, error) {
+	req, err := c.BuildRequest(payload, endpoint, verb)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusAccepted {
+		return nil, fmt.Errorf("error calling KEB: %s %s", resp.Status, resp.Status)
+	}
+
+	var result map[string]interface{}
+	json.NewDecoder(resp.Body).Decode(&result)
+	return result, nil
+}
+
+func (c *KEBClient) CallKEBWithoutToken(payload interface{}, endpoint, verb string) error {
+	req, err := c.BuildRequestWithoutToken(payload, endpoint, verb)
+	fmt.Printf("Request: %s %s\n", req.Method, req.URL)
+	if err != nil {
+		return err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Response:", string(body))
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusForbidden && resp.StatusCode != http.StatusUnauthorized {
+		return fmt.Errorf("unexpected status: %d", resp.StatusCode)
+	}
+	return nil
+}
+
+func (c *KEBClient) GetSKR(instanceID string) (map[string]interface{}, error) {
```

Comment:
Like `KEB` `SKR` is a name we use on daily basis but still it is kind of cryptic.
I do not know if there is better option. Could we try to invent something?

Reply:
We can discuss it with the team today.

Reply:
We decided to change KEB to broker and SKR to instance.